### PR TITLE
added event value and experience change event

### PR DESCRIPTION
**Pull request by GodModed**
### Description
Added the PlayerExpChangeEvent to skript!

---
**Target Minecraft Versions:** 1.14+
**Requirements:** None
**Related Issues:** https://github.com/SkriptLang/Skript/issues/2858

### CODE REVIEW

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 1]
```diff
@@ -1532,5 +1534,16 @@ public Location get(LootGenerateEvent event) {
 				}
 			}, EventValues.TIME_NOW);
 		}
+
+		// PlayerExpChangeEvent
+		if (Skript.classExists("org.bukkit.event.player.PlayerExpChangeEvent")) {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:1539]: You can register your event values in your custom event class

File: src/main/java/ch/njol/skript/events/EvtExperienceChange.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	public boolean check(Event event) {
+		if (mode == ANY)
+			return true;
+		PlayerExpChangeEvent expChangeEvent = (PlayerExpChangeEvent) event;
+		return mode == UP ? expChangeEvent.getAmount() > 0 : expChangeEvent.getAmount() < 0;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "player level progress change " + (mode == UP ? "increase" : "decrease");
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/events/EvtExperienceChange.java:68]: this should account for when `mode == ANY`

**Review by APickledWalrus — CHANGES_REQUESTED** [review]

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]
PlayerExpChangeEvent has been around since 1.7.10, why is it being 1.14 only?

**Comment by GodModed** [comment]
https://github.com/SkriptLang/Skript/pull/5389#pullrequestreview-1264710696
Blame _tud

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/events/EvtExperienceChange.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	public boolean check(Event event) {
+		if (mode == ANY)
+			return true;
+		PlayerExpChangeEvent expChangeEvent = (PlayerExpChangeEvent) event;
+		return mode == UP ? expChangeEvent.getAmount() > 0 : expChangeEvent.getAmount() < 0;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "player level progress change " + (mode == UP ? "increase" : mode == DOWN ? "decrease" : "");
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/events/EvtExperienceChange.java:75]:
```suggestion
		return "player level progress " + (mode == ANY ? "change" : mode == UP ? "increase" : mode == DOWN ? "decrease" : "");
```

**Review by UnderscoreTud — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/events/EvtExperienceChange.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	public boolean check(Event event) {
+		if (mode == ANY)
+			return true;
+		PlayerExpChangeEvent expChangeEvent = (PlayerExpChangeEvent) event;
+		return mode == UP ? expChangeEvent.getAmount() > 0 : expChangeEvent.getAmount() < 0;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "player level progress " + (mode == ANY ? "change" : mode == UP ? "increase" : mode == DOWN ? "decrease" : "");
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/events/EvtExperienceChange.java:75]:
```suggestion
		return "player level progress " + (mode == ANY ? "change" : mode == UP ? "increase" : "decrease");
```
can be simplified

File: src/main/java/ch/njol/skript/events/EvtExperienceChange.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import org.eclipse.jdt.annotation.Nullable;
+
+public class EvtExperienceChange extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("Experience Change", EvtExperienceChange.class, PlayerExpChangeEvent.class, "[player] level progress (change|update|:increase|:decrease)")
+			.description("Called when a player's experience changes.")
+			.examples(
+				"on level progress change:",
+					"\tset {_xp} to event-exp",
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/events/EvtExperienceChange.java:39]: I think the classinfo only supports using `experience` or `experience points` (meaning `event-experience`)

File: src/main/java/ch/njol/skript/events/EvtExperienceChange.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+import ch.njol.skript.util.Experience;
+import ch.njol.skript.util.Getter;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerExpChangeEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+public class EvtExperienceChange extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("Experience Change", EvtExperienceChange.class, PlayerExpChangeEvent.class, "[player] level progress (change|update|:increase|:decrease)")
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/events/EvtExperienceChange.java:35]:
```suggestion
		Skript.registerEvent("Experience Change", EvtExperienceChange.class, PlayerExpChangeEvent.class, "[player] (level progress|[e]xp|experience) (change|update|:increase|:decrease)")
```
I think allowing `on experience increase` is good

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
I think it would be good for you to add support for this event to https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/expressions/ExprExperience.java (it would also mean the amount can be modified)

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 7]
```diff
@@ -50,15 +51,15 @@
...
 public class ExprExperience extends SimpleExpression<Experience> {
 	static {
 		Skript.registerExpression(ExprExperience.class, Experience.class, ExpressionType.SIMPLE, "[the] (spawned|dropped|) [e]xp[erience] [orb[s]]");
 	}
 	
 	@Override
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		if (!getParser().isCurrentEvent(ExperienceSpawnEvent.class, BlockBreakEvent.class)) {
+		if (!getParser().isCurrentEvent(ExperienceSpawnEvent.class, BlockBreakEvent.class, PlayerExpChangeEvent.class)) {
 			Skript.error("The experience expression can only be used in experience spawn and block break events");
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:63]:
```suggestion
			Skript.error("The experience expression can only be used in experience spawn, block break and player experience change events");
```

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 8]
```diff
@@ -72,6 +73,8 @@ protected Experience[] get(final Event e) {
 			return new Experience[] {new Experience(((ExperienceSpawnEvent) e).getSpawnedXP())};
 		else if (e instanceof BlockBreakEvent)
 			return new Experience[] {new Experience(((BlockBreakEvent) e).getExpToDrop())};
+		else if (e instanceof PlayerExpChangeEvent)
+			return new Experience[] {new Experience(((PlayerExpChangeEvent) e).getAmount())};
 		else
 			return new Experience[0];
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:79]: Chained if statements should have curly braces

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 9]
```diff
@@ -100,6 +103,8 @@ public void change(final Event e, final @Nullable Object[] delta, final ChangeMo
 			d = ((ExperienceSpawnEvent) e).getSpawnedXP();
 		else if (e instanceof BlockBreakEvent)
 			d = ((BlockBreakEvent) e).getExpToDrop();
+		else if (e instanceof PlayerExpChangeEvent)
+			d = ((PlayerExpChangeEvent) e).getAmount();
 		else
 			return;
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:109]: Same as above ^

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 10]
```diff
@@ -129,6 +134,8 @@ else if (e instanceof BlockBreakEvent)
 		d = Math.max(0, Math.round(d));
 		if (e instanceof ExperienceSpawnEvent)
 			((ExperienceSpawnEvent) e).setSpawnedXP((int) d);
+		if (e instanceof PlayerExpChangeEvent)
+			((PlayerExpChangeEvent) e).setAmount((int) d);
 		else
 			((BlockBreakEvent) e).setExpToDrop((int) d);
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:140]: Same as above ^

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 11]
```diff
@@ -129,6 +134,8 @@ else if (e instanceof BlockBreakEvent)
 		d = Math.max(0, Math.round(d));
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:134]: Perhaps you can cast this into an `int` here rather than when you're using it

**Review by UnderscoreTud — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/events/EvtExperienceChange.java [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+			@Override
+			@Nullable
+			public Experience get(PlayerExpChangeEvent event) {
+				return new Experience(event.getAmount());
+			}
+		}, EventValues.TIME_NOW);
+	}
+
+	private static final int ANY = 0, UP = 1, DOWN = 2;
+	private int mode = ANY;
```
Comment (kiip1) [inline src/main/java/ch/njol/skript/events/EvtExperienceChange.java:53]: Enum?

Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/events/EvtExperienceChange.java:53]: Probably not needed for such simple use case

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 13]
```diff
@@ -50,16 +51,16 @@
 		"\tif tool of player = diamond pickaxe:",
 		"\t\tadd 100 to dropped experience"})
 @Since("2.1, 2.5.3 (block break event)")
-@Events({"experience spawn", "break / mine"})
+@Events({"experience spawn", "break / mine", "player experience change"})
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:54]:
```suggestion
@Events({"experience spawn", "break / mine", "experience change"})
```

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 14]
```diff
@@ -50,16 +51,16 @@
...
 public class ExprExperience extends SimpleExpression<Experience> {
 	static {
 		Skript.registerExpression(ExprExperience.class, Experience.class, ExpressionType.SIMPLE, "[the] (spawned|dropped|) [e]xp[erience] [orb[s]]");
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:57]: _Just gonna leave this here but it's not related to this PR._
This can be changed in another PR as it allows `xperience` to be used (breaking change)

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java (hunk 8 above)
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:79]: I think this whole class needed a refreshment therefore it can stay as is for now and in another PR the whole classes gets updated

Or if the author doesn't mind they can do it in this PR

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
Nice PR ⚡

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java (hunk 8 above)
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:79]: I think this whole class needed a refreshment therefore it can stay as is for now and in another PR the whole classes gets updated

Or if the author doesn't mind they can do it in this PR

File: src/main/java/ch/njol/skript/events/EvtExperienceChange.java (hunk 12 above)
Comment (TheLimeGlass) [reply src/main/java/ch/njol/skript/events/EvtExperienceChange.java:53]: Agree with Ali

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 15]
```diff
@@ -96,41 +99,44 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
-					case REMOVE_ALL:
-						d -= v;
-						break;
-					case RESET:
-					case DELETE:
-						assert false;
-						break;
-				}
+		}
+		for (final Object o : delta) {
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:111]:
```suggestion
		for (Object obj : delta) {
```

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 16]
```diff
@@ -96,41 +99,44 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
+			final double v = o instanceof Experience ? ((Experience) o).getXP() : ((Number) o).doubleValue();
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:112]:
```suggestion
			double value = obj instanceof Experience ? ((Experience) obj).getXP() : ((Number) obj).doubleValue();
```

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 17]
```diff
@@ -96,41 +99,44 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
 	@Override
 	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
 		double d;
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:101]: This should have a better name

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
Good so far, just minor formatting stuff since you're using them

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java (hunk 17 above)
Comment (GodModed) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:101]: what should the variable be called?

Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:101]: Can be something like eventExp

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 18]
```diff
@@ -95,42 +98,45 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
-			d = 0;
+		}
 		
-		d = Math.max(0, Math.round(d));
-		if (e instanceof ExperienceSpawnEvent)
-			((ExperienceSpawnEvent) e).setSpawnedXP((int) d);
-		else
-			((BlockBreakEvent) e).setExpToDrop((int) d);
+		eventExp = Math.max(0, Math.round(eventExp));
+		int i = (int) eventExp;
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:132]:
```suggestion
		int roundedEventExp = (int) eventExp;
```

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 19]
```diff
@@ -95,42 +98,45 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
+		if (e instanceof ExperienceSpawnEvent) {
+			((ExperienceSpawnEvent) e).setSpawnedXP(i);
+		} else if (e instanceof BlockBreakEvent)
+			((BlockBreakEvent) e).setExpToDrop(i);
+		else {
+			((PlayerExpChangeEvent) e).setAmount(i);
+		}
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:139]:
```suggestion
		if (e instanceof ExperienceSpawnEvent) {
			((ExperienceSpawnEvent) e).setSpawnedXP(i);
		} else if (e instanceof BlockBreakEvent) {
			((BlockBreakEvent) e).setExpToDrop(i);
		} else if (e instanceof PlayerExpChangeEvent) {
			((PlayerExpChangeEvent) e).setAmount(i);
		}
```

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
Almost there ⚡

**Review by AyhamAl-Ali — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 20]
```diff
@@ -95,42 +98,45 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
-			((ExperienceSpawnEvent) e).setSpawnedXP((int) d);
-		else
-			((BlockBreakEvent) e).setExpToDrop((int) d);
+		eventExp = Math.max(0, Math.round(eventExp));
+		int roundedEventExp = (int) eventExp;
+		if (e instanceof ExperienceSpawnEvent) {
+			((ExperienceSpawnEvent) e).setSpawnedXP(roundedEventExp);
+		} else if (e instanceof BlockBreakEvent) {
+			((BlockBreakEvent) e).setExpToDrop(roundedEventExp);
+		} else {
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:137]: This was edited then removed again, it should check for instance before casting
` if (e instanceof PlayerExpChangeEvent)`

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java (hunk 20 above)
Comment (GodModed) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:137]: ok

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 21]
```diff
@@ -95,42 +98,45 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
+			double value = obj instanceof Experience ? ((Experience) obj).getXP() : ((Number) obj).doubleValue();
+			switch (mode) {
+				case ADD:
+					eventExp += value;
+					break;
+				case SET:
+					eventExp = value;
+					break;
+				case REMOVE:
+				case REMOVE_ALL:
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:121]: Basically `REMOVE_ALL` shouldn't be support but since it isn't from your PR, I will just ignore this to avoid Breaking Changes but I think it should be supported

**Review by AyhamAl-Ali — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 22]
```diff
@@ -95,42 +98,45 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
+		if (e instanceof ExperienceSpawnEvent) {
+			eventExp = ((ExperienceSpawnEvent) e).getSpawnedXP();
+		} else if (e instanceof BlockBreakEvent) {
+			eventExp = ((BlockBreakEvent) e).getExpToDrop();
+		} else if (e instanceof PlayerExpChangeEvent) {
+			eventExp = ((PlayerExpChangeEvent) e).getAmount();
+		} else {
 			return;
-		
-		if (delta != null)
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:106]: This check should be restored (likely at the beginning of the method) to avoid any useless work (e.g. setting the exp amount to the same number when delta is null)

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 23]
```diff
@@ -95,42 +98,45 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
+				case REMOVE_ALL:
+					eventExp -= value;
+					break;
+				case RESET:
+				case DELETE:
+					assert false;
+					break;
 			}
-		else
-			d = 0;
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:127]: This can't be removed as it's how the `DELETE` changer works.
For `DELETE`, delta is null so this would originally set the new exp amount to 0. As far as I can tell, `DELETE` would now result in no actual update to the amount - let me know if I'm mistaken though :)

**Review by APickledWalrus — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java (hunk 23 above)
Comment (GodModed) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:127]: Fix it yourself

Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:127]: I await your changes :)

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 24]
```diff
@@ -50,16 +51,16 @@
 		"\tif tool of player = diamond pickaxe:",
 		"\t\tadd 100 to dropped experience"})
 @Since("2.1, 2.5.3 (block break event)")
```
Comment (DelayedGaming) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:53]: `@Since(2.1, 2.5.3 (block break event), INSERT VERSION (experience change event)`

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 25]
```diff
@@ -72,6 +73,8 @@ protected Experience[] get(final Event e) {
 			return new Experience[] {new Experience(((ExperienceSpawnEvent) e).getSpawnedXP())};
 		else if (e instanceof BlockBreakEvent)
```
Comment (DelayedGaming) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:74]: Unable to comment at L71, refractor `e` to `event`

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 26]
```diff
@@ -95,42 +98,45 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
 	
 	@Override
 	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
```
Comment (DelayedGaming) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:100]: Refractor `e` to `event`

**Review by DelayedGaming — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java (hunk 25 above)
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:74]: I think this is outside the scope of this PR. See [comment](https://github.com/SkriptLang/Skript/pull/5389#discussion_r1084344129)

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java (hunk 26 above)
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:100]: I think this is outside the scope of this PR. See [comment](https://github.com/SkriptLang/Skript/pull/5389#discussion_r1084344129)

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java (hunk 23 above)
Comment (GodModed) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:127]: I made the changes.

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 27]
```diff
@@ -95,42 +98,50 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
+			eventExp = ((PlayerExpChangeEvent) e).getAmount();
+		} else {
 			return;
-		
-		if (delta != null)
-			for (final Object o : delta) {
-				final double v = o instanceof Experience ? ((Experience) o).getXP() : ((Number) o).doubleValue();
+		}
+		if (delta == null) {
+			eventExp = eventExp = 0;
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:112]:
```suggestion
			eventExp = 0;
```

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 28]
```diff
@@ -95,42 +98,50 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
...
+		} else if (e instanceof PlayerExpChangeEvent) {
+			eventExp = ((PlayerExpChangeEvent) e).getAmount();
+		} else {
 			return;
-		
-		if (delta != null)
-			for (final Object o : delta) {
-				final double v = o instanceof Experience ? ((Experience) o).getXP() : ((Number) o).doubleValue();
+		}
+		if (delta == null) {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:111]: This may cause unexpected behavior is `delta` is null but the change mode is something like `SET`
I would add something like this as to the beginning of the method (before anything else):
```java
if (delta == null && mode != ChangeMode.DELETE)
	return;
```

**Review by APickledWalrus — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java (hunk 28 above)
Comment (UnderscoreTud) [reply src/main/java/ch/njol/skript/expressions/ExprExperience.java:111]: `delta` can't be null if the mode is `SET`, it calls `change` with `DELETE` instead.

As shown here:
image

**Review by UnderscoreTud — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]

**Comment by GodModed** [comment]
@APickledWalrus @AyhamAl-Ali @TheLimeGlass merge

**Comment by AyhamAl-Ali** [comment]
@GodModed 
Please do not ping the team to merge your PRs. We'll merge the PRs when we think it's ready to be merged.

**Review by TheLimeGlass — APPROVED** [review]
